micro_pumas_utils.F90

```diff
@@ -1341,7 +1342,79 @@ subroutine ice_deposition_sublimation_mg4(t, qv, qi, niic, &
 
   !$acc end data
 end subroutine ice_deposition_sublimation_mg4
-
+            
+subroutine vapor_deposition_onto_snow(t, qv, qs, ns, &
+   precip_frac, rho, dv,qvl, qvi, asn, mu, sc, &
+   vap_deps, vlen)
+
+!INPUT VARS:
+!===============================================
+integer,  intent(in) :: vlen
+real(r8), dimension(vlen), intent(in) :: t
+real(r8), dimension(vlen), intent(in) :: qv
+real(r8), dimension(vlen), intent(in) :: qs
+real(r8), dimension(vlen), intent(in) :: ns
+real(r8), dimension(vlen), intent(in) :: precip_frac
+real(r8), dimension(vlen), intent(in) :: rho
+real(r8), dimension(vlen), intent(in) :: dv
+real(r8), dimension(vlen), intent(in) :: qvl
+real(r8), dimension(vlen), intent(in) :: qvi
+real(r8), dimension(vlen), intent(in) :: asn   ! fall speed parameter for snow
+real(r8), dimension(vlen), intent(in) :: mu
+real(r8), dimension(vlen), intent(in) :: sc
+
+!OUTPUT VARS:
+!===============================================
+real(r8), dimension(vlen), intent(out) :: vap_deps !vapor deposition onto snow (cell-ave value)
+
+!INTERNAL VARS:
+!===============================================
+real(r8) :: ab
+real(r8) :: eps
+real(r8) :: qsic
+real(r8) :: nsic
+real(r8) :: lams
+real(r8) :: n0s
+integer :: i
+
+do i=1,vlen
```

Comment:
need add directives here

Reply:
Okay, added (copied another routine to do this)